File: archive/2020/fredrik/D1/part2.rb
list = File.read("data.txt").split

list.each do |i|
    list.each do |j|
        list.each do |k|
            if i.to_i + j.to_i + k.to_i == 2020
                puts "#{i} #{j} #{k} #{i.to_i*j.to_i*k.to_i}"
            end
        end
    end
end

File: archive/2023/radres/6/part2.py
from helpers import *
with open("input.txt", "r") as f:
    out = f.read()

lines = out.splitlines()
# lines = """Time:      7  15   30
# Distance:  9  40  200""".splitlines()
time = []
dist = []

for c in lines[0].split(":")[1]:
    if c.isdigit():
        time.append(c)
time = int("".join(time))

for c in lines[1].split(":")[1]:
    if c.isdigit():
        dist.append(c)
dist = int("".join(dist))

total_ways = 1
for race in [(time,dist)]:
    speed = 1
    can_beat = False
    time, dist = race
    ways = 0
    while speed < time:
        if dist/speed < (time-speed):
            ways += 1
        speed += 1

    total_ways *= ways

print(total_ways)
